### PR TITLE
Fail the checkpoint if the application fails to start

### LINKED ISSUE #28280: If an application fails to start when doing a checkpoint the checkpoint still succeeds
**Describe the bug**  
If an application fails to start while doing a checkpoint `afterAppStart` the the checkpoint is still allowed to succeed and we are left with a application process checkpoint that will restore always with a failed application start.  It would be better to fail the checkpoint in this case so that container image builds can reliably fail to build if the application fails to start when we are doing a container image build with InstantOn

**Steps to Reproduce**  
Configure any application that fails to start and do a checkpoint with `afterAppStart`

**Expected behavior**

### PR DESCRIPTION
**Pull request by tjwatson**
Fixes #28280

### CODE REVIEW

**Comment by tjwatson** [comment]
#build
fixes #28280

**Comment by tjwatson** [comment]
#libby

File: dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootFailStart.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        server.setCheckpoint(checkpointInfo);
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    @ExpectedFFDC("io.openliberty.checkpoint.internal.criu.CheckpointFailedException")
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootFailStart.java:74]: Should we expect more FFDCs as in the other test?

**Review by anjumfatima90 — APPROVED** [review]
Looks good!

**Pull request merged**